mmda/predictors/hf_predictors/utils.py

```diff
@@ -15,15 +15,16 @@ def shift_index_sequence_to_zero_start(sequence):
 
 
 def get_visual_group_id(token: SpanGroup, field_name: str, defaults=-1) -> int:
-
+    if not hasattr(token, field_name):
+        return defaults
     field_value = getattr(token, field_name)
     if len(field_value) == 0 or field_value[0].id is None:
         return defaults
     return field_value[0].id
 
 
 def convert_document_page_to_pdf_dict(
-    document: Document, page_width: int, page_height: int
+    document: Document
```

Comment:
I got rid of the bbox-coordinate scaling in this method. For one thing, the coordinates were absolute to begin with in my case. For another thing, I was using LayoutLM model that doesn't use the bounding boxes anyway, so loading the page dimensions is a waste.

If confusion between relative and absolute coordinates becomes a problem, I recommend we add a flag to the `Box` class to indicate which coordinates are used in that instance.

Reply:
I don't think we should remove this for two reasons: 

1. the image sizes will not only be used for correcting the block sizes, but also modifying "correcting" token sizes. For original LayoutLM models, it requires to scale all the `token` bounding box to [0, 1000] (which we haven't implemented yet (should be in line 56 below). (for our model, we just use the original coordinates for convenience). 
2. more broadly, this should be a model-specific parameter, rather than be stored in the document object.